lib/Amazon/S3/Thin/SignerV2.pm

```diff
@@ -54,21 +65,30 @@ sub calculate_signature {
     # don't include anything after the first ? in the resource...
     $path =~ /^([^?]*)/;
     $string_to_sign .= "/$1";
+    $path = substr($path, length $1);
```

Comment:
`length $1` can be zero.  so it would be better to return earlier in such a case.

e.g.
```perl
if ($path !~ /^([^?]*)/) {
    ....
    return $string_to_sign;
}